layout/new.go

```diff
@@ -58,6 +63,67 @@ func NewImage(path string, ops ...imgutil.ImageOption) (*Image, error) {
 	}, nil
 }
 
+// NewIndex will return an OCI ImageIndex saved on disk using OCI media Types. It can be modified and saved to a registry
+func NewIndex(repoName, path string, ops ...imgutil.IndexOption) (idx *ImageIndex, err error) {
+	var mfest *v1.IndexManifest
+	var idxOps = &imgutil.IndexOptions{}
+	for _, op := range ops {
+		if err = op(idxOps); err != nil {
+			return idx, err
+		}
+	}
+
+	if err = validateRepoName(repoName, idxOps); err != nil {
+		return idx, err
+	}
+
+	if idxOps.BaseIndex == nil && idxOps.BaseImageIndexRepoName != "" {
+		idxOps.BaseIndex, err = newImageIndexFromPath(idxOps.BaseImageIndexRepoName)
+		if err != nil {
+			return idx, err
+		}
+
+		if idxOps.BaseIndex != nil {
+			// TODO Do we need to do this?
+			mfest, err = idxOps.BaseIndex.IndexManifest()
+			if err != nil {
+				return idx, err
+			}
+
+			if mfest == nil {
+				return idx, errors.New("encountered unexpected error while parsing image: manifest or index manifest is nil")
+			}
+		}
+	}
+
+	localPath := filepath.Join(path, imgutil.MakeFileSafeName(repoName))
+	if idxOps.BaseIndex == nil {
+		if imageExists(localPath) {
+			return idx, errors.Errorf("an image index already exists at %s use FromBaseImageIndex or "+
+				"FromBaseImageIndexInstance options to create a new instance", localPath)
+		}
+	}
+
+	if idxOps.BaseIndex == nil {
+		switch idxOps.Format {
+		case types.DockerManifestList:
+			idxOps.BaseIndex = imgutil.NewEmptyDockerIndex()
+		default:
+			idxOps.BaseIndex = empty.Index
+		}
+	}
```

Comment:
This logic can probably go in the CNBIndex constructor

Reply:
Partially resolved by https://github.com/husni-faiz/imgutil/pull/12 (changed the signature)